spring-cloud-deployer-kubernetes/src/test/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesAppDeployerTests.java

```diff
@@ -139,6 +143,24 @@ public void deployWithVolumesAndVolumeMounts() throws Exception {
                 new VolumeBuilder().withName("testpvc").withNewPersistentVolumeClaim("testClaim", true).build(),
                 new VolumeBuilder().withName("testnfs").withNewNfs("/test/override/nfs", null, "192.168.1.1:111").build());
     }
+    @Test
+    public void deployWithVolumesAndVolumeMountsOnAdditionalContainer() throws Exception {
+        AppDefinition definition = new AppDefinition("app-test", null);
+        Map<String, String> props = new HashMap<>();
+        props.put("spring.cloud.deployer.kubernetes.volumes", "[{name: 'config', configMap: {name: promtail-config, items: [{key: promtail.yaml, path: promtail.yaml}]}}]");
```

Comment:
Nitpick: Could we add more volumes so that we can make sure the correct one is returned?

Reply:
done